validator/service/service.go

```diff
@@ -97,6 +97,29 @@ type CheckBlobResult struct {
 	MismatchedData []string
 }
 
+// shouldRetry returns true if the status code is one of the retryable status codes
+func shouldRetry(status int) bool {
+	switch status {
+	case http.StatusInternalServerError, http.StatusBadGateway, http.StatusServiceUnavailable, http.StatusGatewayTimeout, http.StatusTooManyRequests:
+		return true
+	default:
+		return false
+	}
+}
+
+// fetchWithRetries fetches the sidecar and handles retryable error cases (5xx status codes + 429 + connection errors)
+func fetchWithRetries(ctx context.Context, endpoint BlobSidecarClient, id string, format Format) (int, storage.BlobSidecars, error) {
+	return retry.Do2(ctx, retryAttempts, retry.Exponential(), func() (int, storage.BlobSidecars, error) {
+		status, resp, err := endpoint.FetchSidecars(id, format)
+
+		if err == nil && status != http.StatusOK && shouldRetry(status) {
```

Comment:
should you retry on (certain) errors as well?  I can see occasional connectivity issues causing errors to be returned.

Reply:
I believe this will retry when `endpoint.FetchSidecars` returns a error (due to a connectivity issue). In that case err != nil, and it'll just fall through to return the original error (which triggers the retry).

Adding this if statement, just ensures that err is set (and will trigger a retry) when the status code is a 5xx or 429.

If you think it's cleaner, we could change this to:

```
if shouldRetry(status) { return 0, BlobSidecars{}, errors.New(...) }
```

Reply:
ah I see, no it's fine as is, should have taken a bit more time to understand what was going on.